hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java

```diff
@@ -351,23 +351,15 @@ private Pair<Boolean, List<CleanFileInfo>> getFilesToCleanKeepingLatestCommits(S
             continue;
           }
 
-          if (policy == HoodieCleaningPolicy.KEEP_LATEST_COMMITS) {
+          if (policy == HoodieCleaningPolicy.KEEP_LATEST_COMMITS || policy == HoodieCleaningPolicy.KEEP_LATEST_BY_HOURS) {
```

Comment:
If we agree this is the correct behavior, I can update or add unit tests

Reply:
this logic is only applicable to these 2 policies. we can validate the policy at the beginning, and save the checks here?

Reply:
sure, to be clear do you mean add a precondition or something like that at the beginning of the method?